tests/auth_tests/test_middleware.py

```diff
@@ -43,3 +45,17 @@ def test_no_session(self):
         )
         with self.assertRaisesMessage(ImproperlyConfigured, msg):
             self.middleware(HttpRequest())
+
+    async def test_auser_then_user(self):
+        self.middleware(self.request)
+        self.assertIsNotNone(await self.request.auser())
+        # Note that since `auser` and `user` use the same cache
+        # the following will work in an async function AFTER
+        # `auser` has been called
+        self.assertIsNotNone(self.request.user)
+        self.assertFalse(self.request.user.is_anonymous)
+
+    async def test_user_then_auser(self):
+        self.middleware(self.request)
+        self.assertIsNotNone(await sync_to_async(lambda: self.request.user)())
+        self.assertFalse((await self.request.auser()).is_anonymous)
```

Comment:
How many DB queries are executed here? 🤔

Reply:
Not sure this is relevant with the changes I'm about to push, but it was only 1 (L61 hits the cached user created by L60)